File: templates/templates/EpoxyHello.Uwp/EpoxyHello/Views/Converters/ScoreToBrushConverter.cs
////////////////////////////////////////////////////////////////////////////
//
// Epoxy template source code.
// Write your own copyright and note.
// (You can use https://github.com/rubicon-oss/LicenseHeaderManager)
//
////////////////////////////////////////////////////////////////////////////

using Epoxy;
using Windows.UI;
using Windows.UI.Xaml.Media;

namespace EpoxyHello.Views.Converters
{
    public sealed class ScoreToBrushConverter : ValueConverter<int, Brush>
    {
        private static readonly Brush yellow = new SolidColorBrush(Color.FromArgb(255, 96, 96, 0));
        private static readonly Brush gray = new SolidColorBrush(Color.FromArgb(255, 96, 96, 96));

        public override bool TryConvert(int from, out Brush result)
        {
            result = from >= 5 ? yellow : gray;
            return true;
        }
    }
}

File: templates/templates/FSharp.EpoxyHello.Avalonia/EpoxyHello.UI/Views/Converters/ScoreToBrushConverter.fs
////////////////////////////////////////////////////////////////////////////
//
// Epoxy template source code.
// Write your own copyright and note.
// (You can use https://github.com/rubicon-oss/LicenseHeaderManager)
//
////////////////////////////////////////////////////////////////////////////

namespace EpoxyHello.Views.Converters

open Epoxy
open Avalonia.Media

[<Sealed>]
type public ScoreToBrushConverter() =
    inherit ValueConverter<int, Brush>()

    let yellow = new SolidColorBrush(Color.FromArgb(255uy, 96uy, 96uy, 0uy)) :> Brush
    let gray = new SolidColorBrush(Color.FromArgb(255uy, 96uy, 96uy, 96uy)) :> Brush

    override __.convert from =
        if from >= 5 then Some yellow else Some gray
